netbox/dcim/forms/common.py

```diff
@@ -52,8 +56,10 @@ def clean(self):
             })
 
         # Remove all tagged VLAN assignments from "tagged all" interfaces
-        elif self.cleaned_data['mode'] == InterfaceModeChoices.MODE_TAGGED_ALL:
-            self.cleaned_data['tagged_vlans'] = []
+        elif self.cleaned_data['mode'] == InterfaceModeChoices.MODE_TAGGED_ALL and tagged_vlans:
+            raise forms.ValidationError({
+                'mode': _("An tagged-all interface cannot have tagged VLANs assigned.")
```

Comment:
"An" should be "A"

Reply:
Should this be the same logic as above for the API?  In the API case you are checking for  `mode != InterfaceModeChoices.MODE_TAGGED` and here you are testing for `self.cleaned_data['mode'] == InterfaceModeChoices.MODE_TAGGED_ALL`

Reply:
Funny enough, after reviewing this PR I realised that the invalid data is just removed in __init__ because we delete that item.

I am not sure if this is the best way to do this IMO.